app/models/user.rb

```diff
@@ -17,6 +18,30 @@ class User < ApplicationRecord
     presence: true,
     length: {minimum: Settings.minimum_password_length}
 
+  def self.digest string
+    min_cost = ActiveModel::SecurePassword.min_cost
+    cost = min_cost ? BCrypt::Engine::MIN_COST : BCrypt::Engine.cost
+    BCrypt::Password.create string, cost: cost
+  end
+
+  def self.new_token
+    SecureRandom.urlsafe_base64
+  end
+
+  def remember
+    self.remember_token = User.new_token
+    update_attribute :remember_digest, User.digest(remember_token)
+  end
+
+  def authenticated? remember_token
+    return nil if remember_digest.nil?
+    BCrypt::Password.new(remember_digest).is_password? remember_token
+  end
+
+  def forget
+    update_attribute :remember_digest, nil
```

Comment:
cái update_attribute này có đảm bảo toàn vẹn dữ liệu không em nhỉ :-?

Reply:
theo em nghĩ là không, vì update_attribute sẽ bỏ qua validate